### ISSUE TITLE
FN S3981: No issue is raised when property check is done through a pattern.

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5983]
### Description

No issue is raised when property check is done through a pattern.

### Repro steps

```csharp
using System;
using System.Linq;

var r = new R();

if (r is R { SomeProperty.Length: >= 0 }) // FN
{
}
```

### Related information

* C#/VB.NET Plugins version: 8.43.0.51858
